ros2_batch_job/packaging.py

```diff
@@ -112,30 +112,31 @@ def build_and_test_and_package(args, job):
     # Only on Linux and OSX Python scripts have a shebang line
     if args.os in ['linux', 'osx']:
         print('# BEGIN SUBSECTION: rewrite shebang lines')
-        bin_path = os.path.join(args.installspace, 'bin')
-        for filename in os.listdir(bin_path):
-            path = os.path.join(bin_path, filename)
-            with open(path, 'rb') as h:
-                content = h.read()
-            shebang = b'#!%b' % job.python.encode()
-            if content[0:len(shebang)] != shebang:
-                continue
-            print('- %s' % path)
-            with open(path, 'wb') as h:
-                h.write(b'#!/usr/bin/env python3')
-                h.write(content[len(shebang):])
+        # Demo nodes are installed to 'lib', so binaries may be present there in addition to 'bin'
+        for dir_with_binaries in ['bin', 'lib']:
+            path_with_binaries = os.path.join(args.installspace, dir_with_binaries)
+            for path in [os.path.join(root, f) for root, dirs, files in os.walk(path_with_binaries) for f in files]:
```

Comment:
The algorithm should only try to touch very specific files. While we currently don't have files in the other locations there could be some in the future and they shouldn't be changed.